pkg/controller/migration/convert/network.go

```diff
@@ -41,30 +41,40 @@ func handleCalicoCNI(c *components, install *Installation) error {
 
 	errCtx := fmt.Sprintf("detected %s CNI plugin", plugin)
 
-	// CALICO_NETWORKING_BACKEND
-	netBackend, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "CALICO_NETWORKING_BACKEND")
-	if err != nil {
-		return err
-	}
-
-	if netBackend != nil && *netBackend != "" && *netBackend != "bird" {
-		return ErrIncompatibleCluster{fmt.Sprintf("%s, only CALICO_NETWORKING_BACKEND=bird is supported at this time", errCtx)}
-	}
-
-	// Calico CNI
 	if c.calicoCNIConfig == nil {
-		return ErrIncompatibleCluster{fmt.Sprintf("%s, required cni config was not found ", errCtx)}
+		return ErrIncompatibleCluster{fmt.Sprintf("%s, required Calico cni config was not found ", errCtx)}
 	}
 
 	if install.Spec.CNI == nil {
 		install.Spec.CNI = &operatorv1.CNISpec{}
 	}
 	install.Spec.CNI.Type = plugin
 
+	if install.Spec.CNI.IPAM == nil {
```

Comment:
all these nil checks here - is it safe to say that they will all need to be initialized during a migration since we are detecting the existing networking mode? If so, maybe we can do this earlier?

Reply:
I don't think I would want to do this earlier. The CalicoNetwork and IPAM fields may not be set in all cases (like for non-Calico CNI). I believe the CNI one will always be done but I think I'd prefer to keep it close to where we actually set it.